plugins/controlboard/src/ControlBoardDriver.cpp

```diff
@@ -252,7 +252,8 @@ bool ControlBoardDriver::setControlMode(const int j, const int mode)
         return false;
     }
 
-    m_controlBoardData->joints.at(j).controlMode = mode;
+    m_controlBoardData->joints.at(j).controlMode = (mode == VOCAB_CM_FORCE_IDLE) ? VOCAB_CM_IDLE
```

Comment:
I suggest to implement the same logic used in https://github.com/robotology/gazebo-yarp-plugins/blob/d296652b7619dbef39e9bd65ae49f6ca3bdd9377/plugins/controlboard/src/ControlBoardDriverControlMode.cpp#L115-L129

```suggestion
       // If joint is in hw fault, only a force idle command can recover it
    if (m_controlBoardData->joints.at(j).controlMode == VOCAB_CM_HW_FAULT && mode != VOCAB_CM_FORCE_IDLE){
        return true;
    }

    if (mode == VOCAB_CM_FORCE_IDLE)
    {
        // Clean the fault status and set control mode to idle
        mode = VOCAB_CM_IDLE;
    }

    m_controlBoardData->joints.at(j).controlMode = mode;
```

Reply:
awesome